nanofi/include/core/cstructs.h

```diff
@@ -89,8 +89,7 @@ typedef struct processor_context processor_context;
 
 typedef struct {
   const char *key;
-  void *value;
-  size_t value_size;
+  const char *value;
```

Comment:
I would probably leave the size because a) you don't know if the value is null terminated ( it may not be ) -- and b) as we move away from C++ we will likely manage the memory and update_attribute at that point could/may be a realloc.

Reply:
My idea here was to make sure that whatever we store is null-terminated.
We provide API to add non-terminated str with length, (something like strncpy), but inside store null-terminated things.

Two reasons behind:
-no need to store len in internal structures
-whatever we return is null-terminated, which makes it easy to copy/log

In my opinion this leads to a more compact  and cleaner API, but I can be convinced about the opposite way, too.

Reply:
What would convince you?

Reply:
> We provide API to add non-terminated str with length, (something like strncpy), but inside store null-terminated things.

We can not provide non-null-terminated API for null-terminated storage, because the former allows for null bytes in the payload while the latter does not.
+1 for keeping `void*` + `size_t` if the attribute value can ever contain a null byte as part of the payload, -1 if it is always a string that can be null-terminated.